src/online2/online-nnet3-wake-word-faster-decoder.h

```diff
@@ -53,8 +53,8 @@ class OnlineWakeWordFasterDecoder : public FasterDecoder {
  public:
   OnlineWakeWordFasterDecoder(const fst::Fst<fst::StdArc> &fst,
                               const OnlineWakeWordFasterDecoderOpts &opts,
-                              const TransitionModel &trans_model)
-      : FasterDecoder(fst, opts), opts_(opts), trans_model_(trans_model) {}
```

Comment:
I guess the trans_model arg can be deleted, although you might have to change the calling code as well.  if there are no compilation errors, OK to delete.

Reply:
Yes, it goes without saying that calling code also changes.

I'm also wondering if it's worth it to declare this constructor `[[deprecated("Use the 42-argument constructor")]]` for some value of 42 instead of just removing it? I'm cool breaking internal interfaces, but do not like breaking the public API. `OnlineWakeWordFasterDecoder` is certainly useful at inference time, hence in production.

A last one I want to run by you. I noticed in the commit that I merged today: a public constructor
```c++
NnetChainModel2::NnetChainModel2(const NnetChainTraining2Options& /* unused */,
                                 Nnet *nnet, const std::string &den_fst_dir)
    : nnet(nnet), den_fst_dir_(den_fst_dir) {}
```
which became unused when I removed unused member `const NnetChainTraining2Options &opts_` (but the same `NnetChainTraining2Options` to `NnetChainTrainer2` struct is used, and it looks like it's where it should have been from the start. But again, don't want to break a public API. It's unlikely the _trainer_ will be used with anything but own Kaldi binary, but the extra argument is on the `NnetChainModel2`. 

Am I reading the code right, Is this is a training-time only wrapper around `Nnet*`? Then the extra parameter would be ok to remove w/o going through deprecation. Or is it a useful inference-time public class?

Reply:
OnlineWakeWordFasterDecoder  is probably not used enough yet, to worry about breaking APIs.
Sorry I don't recall very much about the NnetChainModel2 code, you'll have to figure it out yourself.  It's likely not very much used right now.

Reply:
Thanks! I won't worry about the OWWFD then, and will figure out the other one, no biggie.